### PR TITLE
PAT-564 Create Pathfinder API and script - migration script

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: server/services/pathfinderService.js [hunk 1]
```diff
@@ -268,14 +274,15 @@ module.exports = function createPathfinderService({ pathfinderClient, elite2Clie
...
 
-  async function createNewRiskBandingAssessment(context, nominalId, username, form, band, userId) {
+  async function createNewRiskBandingAssessment(context, nominalId, username, form, band, userId, date) {
     const id = await pathfinderClient.createRiskBandingAssessment(
       nominalId,
       riskBandingStatus.NEW,
       username,
       form,
       band,
       userId,
```
Comment (steverendell) [inline server/services/pathfinderService.js]: This 1-hour adjustment is a bit horrible but might be just due to my local env. If not needed in dev ill remove it later

File: migrate-linkspace-data.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+SESSION_SIG_COOKIE=xypIEaasmUneFc-vY6UK15QvIKc
+
```
Comment (garethmdavies) [inline migrate-linkspace-data.sh]: Should we default these two variables to empty strings if we are going to replace them during the actual run?

File: migrate-linkspace-data.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+# Check whether the file exists and is readable
+if [[ ! -f "$FILE" ]]; then
+  echo "Unable to find file $FILE"
+  exit 1
+fi
+
+# get the csrf :  <input type="hidden" name="_csrf" value="Se5RABWQ-ptiJb5h-M7r-IoPd5VqfpbZkjoA" />    -H "Content-Type: application/x-www-form-urlencoded"
+# This just needs to use any CRN which exists
+CSRF=$(curl --cookie "session=$SESSION_COOKIE;session.sig=$SESSION_SIG_COOKIE" --cookie-jar cookie-jar $APP_HOST/refer/offender/crn/D008435 | awk '/_csrf/ { print $4 }' | awk -F \" '{print $2}')
+if [ -z "$CSRF" ]; then
```
Comment (garethmdavies) [inline migrate-linkspace-data.sh]: Maybe parameterise the crn which may need to be different for each environment.

File: migrate-linkspace-data.sh [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  if [[ "$id" == "ID" || "$id" == "OFFICIAL SENSITIVE" ]]; then
+    continue
+  fi
+
+  echo "----------------- Processing: ${row[*]}"
+  nomsid=${row[10]}
+  crn=${row[11]}
+  source=NOMIS
+  if [ -z "$nomsid" ]; then
+    source=DELIUS
```
Comment (garethmdavies) [inline migrate-linkspace-data.sh]: What does the -z do here?

Comment (steverendell) [reply migrate-linkspace-data.sh]: its a string empty test

File: migrate-linkspace-data.sh (hunk 2 above)
Comment (steverendell) [reply migrate-linkspace-data.sh]: ok, Ill move these values to comments to help identify whats is needed

File: migrate-linkspace-data.sh (hunk 3 above)
Comment (steverendell) [reply migrate-linkspace-data.sh]: Done

**Pull request merged**